crates/typst/src/math/ctx.rs

```diff
@@ -53,7 +61,7 @@ pub struct MathContext<'a, 'b, 'v> {
     pub ssty_table: Option<ttf_parser::gsub::AlternateSubstitution<'a>>,
     pub glyphwise_tables: Option<Vec<GlyphwiseSubsts<'a>>>,
     pub space_width: Em,
-    pub fragments: Vec<MathFragment>,
+    pub fragments: Vec<(MathFragment, GroupRole)>,
```

Comment:
As discussed in https://github.com/typst/typst/pull/1826#issuecomment-1690324284, the group role thing should be removed here, so that the code becomes cleaner again and breaks in `lr` are allowed.